src/PrestaShopBundle/Command/GenerateHooksDocumentationCommand.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Command;
+
+use Exception;
+use PrestaShop\PrestaShop\Core\Hook\Extractor\HookExtractor;
+use PrestaShop\PrestaShop\Core\Hook\Provider\GridDefinitionHookByServiceIdsProvider;
+use SimpleXMLElement;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+#[AsCommand(
+    name: 'prestashop:update:hooks-documentation',
+    description: 'Extract Hooks Documentation files',
+)]
+final class GenerateHooksDocumentationCommand extends Command
+{
+    protected static $defaultName = 'prestashop:extract:hooks';
+
+    public function __construct(
+        private readonly HookExtractor $hookExtractor,
+        private string $hookFile
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Extract Hooks Documentation files')
+            ->addOption(
+                'output-dir',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Directory to output the generated markdown files, by default it will be docs/hooks/',
+                '/docs/hooks/'
```

Comment:
I don't think we should have this default behaviour, at best it will pollute the local shop At worst people won't know they are supposed to use this to generate/update the doc repository

Since this command is only supposed to be used to update the docs we could force the parameter